### PR TITLE
Store offset and create_key_hash as hex encoded varchar

### PR DESCRIPTION
**Pull request by gerolf-da**
This increases compatibility with Oracle.

### Pull Request Checklist

- [ ] Read and understand the [contribution guidelines](https://github.com/digital-asset/daml/blob/main/CONTRIBUTING.md)
- [ ] Include appropriate tests
- [ ] Set a descriptive title and thorough description
- [ ] Add a reference to the [issue this PR will solve](https://github.com/digital-asset/daml/issues), if appropriate
- [ ] Include changelog additions in one or more commit message bodies between the `CHANGELOG_BEGIN` and `CHANGELOG_END` tags
- [ ] Normal production system change, include purpose of change in description

NOTE: CI is not automatically run on non-members pull-requests for security
reasons. The reviewer will have to comment with `/AzurePipelines run` to
trigger the build.

### CODE REVIEW

**Comment by gerolf-da** [comment]
@dasormeter: I think this could be of interest to you ;)

**Comment by dasormeter** [comment]
> @dasormeter: I think this could be of interest to you ;)

Good stuff - bring on the hexes :)

**Review by rautenrieth-da — COMMENTED** [review]
The change itself looks good, but can you please wait until https://github.com/digital-asset/daml/pull/9394 is merged (which this PR will conflict with), and until we are convinced that this PR will not introduce a regression in the index performance?

File: ledger/participant-integration-api/src/main/scala/platform/indexer/parallel/PostgresDAO.scala
Comment (gerolf-da) [inline ledger/participant-integration-api/src/main/scala/platform/indexer/parallel/PostgresDAO.scala:771]: @mziolekda, @kamil-da: On review, I'm slightly less comfortable with relying on `HexStringModuleImpl#encode` to never ever change in the future. It'll be slightly more future proof to extract the specific serialization into a helper function in the DB access layer. This can be done in a separate PR though.

cc @remyhaemmerle-da, @moritzkiefer-da:

**Review by gerolf-da — COMMENTED** [review]
Thank you for rebasing, @kamil-da! 🙇 
Only left 1 comment about an issue that I'd like a decision on before we merge.

File: ledger/participant-integration-api/src/main/scala/platform/indexer/parallel/PostgresDAO.scala
Comment (kamil-da) [reply ledger/participant-integration-api/src/main/scala/platform/indexer/parallel/PostgresDAO.scala:771]: @gerolf-da @cocreature I don't have strong feelings against using `HexStringModuleImpl` either.

@gerolf-da Is there anything stopping us from merging this once PR checks complete? Any benchmarks?

Comment (gerolf-da) [reply ledger/participant-integration-api/src/main/scala/platform/indexer/parallel/PostgresDAO.scala:771]: No blocker to merge this PR. We can change that, if we deem it necessary, in a subsequent PR. I can also not approve it because I opened it ;)

Comment (kamil-da) [reply ledger/participant-integration-api/src/main/scala/platform/indexer/parallel/PostgresDAO.scala:771]: I understand, thanks!

**Review by kamil-da — APPROVED** [review]

File: ledger/participant-integration-api/src/main/scala/platform/indexer/parallel/PostgresDAO.scala
Comment (remyhaemmerle-da) [reply ledger/participant-integration-api/src/main/scala/platform/indexer/parallel/PostgresDAO.scala:771]: Your choice.

**Pull request merged**
